common/djangoapps/enrollment/views.py

```diff
@@ -166,8 +166,7 @@ def get(self, request, course_id=None, username=None):
         username = username or request.user.username
 
         # TODO Implement proper permissions
-        if request.user.username != username and not self.has_api_key_permissions(request) \
-                and not request.user.is_superuser:
+        if request.user.username != username and not request.user.is_staff:
```

Comment:
I'm not certain we have removed all usages of the API key header. Until then that access permission must remain intact. Revert these lines and simply change is_superuser to is_staff.